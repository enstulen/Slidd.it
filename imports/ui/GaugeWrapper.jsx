import React, { Component, PropTypes } from 'react';
import Gauge from 'react-svg-gauge';

// Wrapper component for the gauge on the app page.
export default class GaugeWrapper extends Component {

  // Initial values:
  // value = Starting value of gauge.
  constructor(props) {
    super(props);
    this.state = {
      value: 0,
      width: 500,
      height: 320,
    };
  }
  // Calculates the avarage of array values. Rounds the number.
  calculateAvarage(array) {
    if (array.length > 0) {
      let total = 0;
      for (let i = 0; i < array.length; i += 1) {
        total += array[i].value;
      }
      const average = total / array.length;
      return Math.round(average);
    }
    return 0;
  }

  render() {
    return (
      <div>
        <Gauge className="gauge" value={this.calculateAvarage(this.props.sliderValues)} width={this.state.width} height={this.state.height} label="" />
      </div>
    );
  }
}

// // // // // PropTypes // // // // //

GaugeWrapper.propTypes = {
  sliderValues: PropTypes.array.isRequired,
};
