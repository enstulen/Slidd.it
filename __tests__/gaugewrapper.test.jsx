import renderer from 'react-test-renderer';
import React from 'react';
import { mount, shallow } from 'enzyme';
import GaugeWrapper from '../imports/ui/GaugeWrapper.jsx';

describe('GaugeWrapper component', () => {
  it('Renders correctly', () => {
    const rendered = renderer.create(
      <GaugeWrapper sliderValues={[1, 4]} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('Correct length', () => {

    const wrapper = shallow(<GaugeWrapper sliderValues={[1]} />);
    expect(wrapper.find('.gauge').length).toEqual(1);
  });
});
