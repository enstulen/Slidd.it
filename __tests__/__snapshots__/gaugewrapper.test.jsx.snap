// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GaugeWrapper component Renders correctly 1`] = `
<div>
  <svg
    height="100%"
    style={
      Object {
        "height": 320,
        "left": 0,
        "overflow": "hidden",
        "position": "relative",
        "top": 0,
        "width": 500,
      }
    }
    version="1.1"
    width="100%"
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <filter
        id="g3-inner-shadow"
      >
        <feOffset
          dx="0"
          dy="3"
        />
        <feGaussianBlur
          result="offset-blur"
          stdDeviation="5"
        />
        <feComposite
          in="SourceGraphic"
          in2="offset-blur"
          operator="out"
          result="inverse"
        />
        <feFlood
          floodColor="black"
          floodOpacity="0.2"
          result="color"
        />
        <feComposite
          in="color"
          in2="inverse"
          operator="in"
          result="shadow"
        />
        <feComposite
          in="shadow"
          in2="SourceGraphic"
          operator="over"
        />
      </filter>
    </defs>
    <path
      d="M125,256 L50,256 A200,200 0 0 1 450,256 L375,256 A125,125 0 0 0 125,256 Z "
      fill="#edebeb"
      filter="url(#g3-inner-shadow)"
      stroke="none"
    />
    <path
      d="M125,256 L50,256 A200,200 0 0 1 NaN,NaN LNaN,NaN A125,125 0 0 0 125,256 Z "
      fill="#fe0400"
      filter="url(#g3-inner-shadow)"
      stroke="none"
    />
    <text
      style={
        Object {
          "fill": "#999999",
          "fillOpacity": 1,
          "fontSize": 50,
          "fontStretch": "normal",
          "fontStyle": "normal",
          "fontVariant": "normal",
          "fontWeight": "bold",
          "lineHeight": "normal",
          "stroke": "none",
          "textAnchor": "middle",
        }
      }
      textAnchor="middle"
      x={250}
      y={40}
    >
      
    </text>
    <text
      style={
        Object {
          "fill": "#010101",
          "fillOpacity": 1,
          "fontSize": 100,
          "fontStretch": "normal",
          "fontStyle": "normal",
          "fontVariant": "normal",
          "fontWeight": "bold",
          "lineHeight": "normal",
          "stroke": "none",
          "textAnchor": "middle",
        }
      }
      textAnchor="middle"
      x={250}
      y={256}
    >
      NaN
    </text>
    <text
      style={
        Object {
          "fill": "#999999",
          "fillOpacity": 1,
          "fontSize": 20,
          "fontStretch": "normal",
          "fontStyle": "normal",
          "fontVariant": "normal",
          "fontWeight": "normal",
          "lineHeight": "normal",
          "stroke": "none",
          "textAnchor": "middle",
        }
      }
      textAnchor="middle"
      x={87.5}
      y={281}
    >
      0
    </text>
    <text
      style={
        Object {
          "fill": "#999999",
          "fillOpacity": 1,
          "fontSize": 20,
          "fontStretch": "normal",
          "fontStyle": "normal",
          "fontVariant": "normal",
          "fontWeight": "normal",
          "lineHeight": "normal",
          "stroke": "none",
          "textAnchor": "middle",
        }
      }
      textAnchor="middle"
      x={412.5}
      y={281}
    >
      100
    </text>
  </svg>
</div>
`;
